assets/app/features/upgrade/controllers/upgrade-database-configuration.controller.js

```diff
@@ -11,20 +11,86 @@
     angular.module('crowbarApp.upgrade')
         .controller('UpgradeDatabaseConfigurationController', UpgradeDatabaseConfigurationController);
 
-    UpgradeDatabaseConfigurationController.$inject = ['upgradeStepsFactory'];
+    UpgradeDatabaseConfigurationController.$inject = ['upgradeStepsFactory', 'upgradeFactory'];
     // @ngInject
-    function UpgradeDatabaseConfigurationController(upgradeStepsFactory) {
+    function UpgradeDatabaseConfigurationController(upgradeStepsFactory, upgradeFactory) {
         var vm = this;
         vm.databaseForm = {
             username: '',
             password: '',
-            server: '',
-            port: 5432,
-            tablePrefix: ''
+            host: '',
+            port: 5432
         };
-        // Set the current step as completed by default to enable the next button
-        // TODO(itxaka): remove this when we have the proper workflow for this step
-        upgradeStepsFactory.setCurrentStepCompleted();
 
+        /* eslint-disable max-len */
+        // NOTE: patterns based on https://github.com/crowbar/crowbar-client/blob/master/lib/crowbar/client/mixin/database.rb
+        vm.patterns = {
+            hostname: '^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$',
+            ipv4: '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$',
+            username: '^[a-zA-Z0-9_]*$',
+            password: "^[a-zA-Z0-9_$&+,:;=?@#|'<>.^*()%!-]*$",
+        };
+        /* eslint-enable max-len */
+        vm.patterns.host = vm.patterns.hostname + '|' + vm.patterns.ipv4;
+
+        vm.running = false;
+        vm.spinnerVisible = false;
```

Comment:
Not sure I understand your question. If you mean that this one doesn't need to be defined in the controller then yes, it can be done implicitly by angular.